src/main/kotlin/org/jitsi/rtp/rtp/RtpSequenceNumber.kt

```diff
@@ -54,6 +54,7 @@ inline class RtpSequenceNumber(val value: Int) : Comparable<RtpSequenceNumber> {
 
     companion object {
         val INVALID = RtpSequenceNumber(-1)
+        fun wrap(value: Int): RtpSequenceNumber = RtpSequenceNumber(value and 0xffff)
```

Comment:
This is a good catch!  I wonder if there's a good way we can enforce the constraint on the inline class so we don't have to remember to call `wrap`.  I'll look into it.

Reply:
I tried to do it in the constructor, but I get `Inline class primary constructor must have only final read-only (val) property parameter`. Maybe we can make it private and provide an `invoke` (or would that defeat the purpose of `inline`?)

Reply:
That was my first thought, too, but inline classes must have a public primary constructor.  I looked at the implementation of `UInt` in kotlin, and I think they actually enforce the unsigned behaviors in the operations, instead of in the underlying value (`compareTo`, etc.)

Reply:
Looks like we actually do this already in some places (`plus`, and `minus`), we just need to do it in `compareTo` and `rangeTo`, I think.

Reply:
What about reading `value` before any operations are performed?

Reply:
Yes that's a good question...I'm curious to know how best to do this.  Give me a little time to look into it.

Reply:
I took a look at the `UInt` implementation and the strategy I found there is this:

1. Make the constructor `internal` (involves suppressing a warning, which `UInt` does)
1. Make the wrapped member `internal`
1. All instances are created via extension functions (such as `Int.toUInt()`), so we'd do that for `RtpSequenceNumber`: `Int.toRtpSequenceNumber`, etc.
1. The internal type is always 'conformed' to what's desired; as in, they hold it as an `Int` but make sure it always behaves as a `UInt`

Now, the `internal` trick solves the "accessing `value` directly" problem, but it's not _quite_ as useful for us, since `rtp` will still be able to access it.  I think we can document the class that the constructor should not be used...but our only other option is to create another inner maven module in there to enforce the `internal` scope even within `rtp`.

This, at least, is where I think we'd want to end up to be most "correct".  We don't necessarily need to do it now, but I think we should use `Int.toRtpSequenceNumber` instead of `RtpSequenceNumber.wrap` as they're functionally the same but the former is where we'd want to end up anyway.

Reply:
I looked at doing this and I'm confused. If the wrapped value is `internal` how is it exposed?

If we are suppressing the warning anyway, we can simply mark the constructor `private`, assure the value is always coerced (in `Int.toRtpSequenceNumber()`), and continue to expose the value. This is what I've done. Let me know if I missed some of your points.


I also tried to delay coercion until the numbers are read by making the value field private and exposing it via `toInt`. This is nice because all operations are performed on the standard ints until we need a result, but it doesn't work with comparison. I don't understand why, but kotlin 
does not use `compareTo` when checking for equality resulting in `RtpSequenceNumber(0) shouldBe RtpSequenceNumber(0x10000)` failing (although <, >, <=, >= work correctly).

Reply:
Cool, I think the way it ended up looks nice.  But are you saying the equality is not working correctly?  Or that was with another way you tried?

Reply:
Equality works now (since we only ever created instances with coerced values). It didn't work with the other approach I tried (which is disappointing).